var/spack/repos/builtin/packages/gdal/package.py

```diff
@@ -708,7 +708,9 @@ def configure_args(self):
             self.with_or_without("xerces", variant="xercesc", package="xerces-c"),
             self.with_or_without("zstd", package="zstd"),
             # Language bindings
-            self.with_or_without("python", package="python", attribute="command"),
+            self.with_or_without(
+                "python", variant="python", package="python-venv", attribute="command"
+            ),
```

Comment:
It would be nice to avoid leaking the knowledge of `python-venv` here.  I suggested above making a method for:

```python
self.spec.dependencies("python-venv") or self.spec.dependencies("python")
```

In `PythonExtension`. I think you could use that here, e.g.:

```suggestion
            self.with_or_without(
                "python", variant="python", package=self.python.name, attribute="command"
            ),
```

That you even have to do that is kind of leaky but I think we shouldn't have packages making assumptions about it.

Reply:
`package="python", attribute="command` works now.